src/browser/vdiff.js

```diff
@@ -1,28 +1,106 @@
-import { chai, expect } from '@open-wc/testing';
+import { chai, expect, nextFrame } from '@open-wc/testing';
 import { executeServerCommand } from '@web/test-runner-commands';
 
-let test;
+let test, soonPromise;
 
-chai.Assertion.addMethod('golden', function(...args) {
-	return ScreenshotAndCompare.call({ test, elem: this._obj }, ...args); // eslint-disable-line no-invalid-this
+/* eslint-disable no-undef, no-invalid-this */
+chai.Assertion.addMethod('golden', async function(...args) {
+	await soonPromise?.catch(err => expect.fail(err));
+	return ScreenshotAndCompare.call({ test, elem: this._obj }, ...args);
 });
-mocha.setup({ // eslint-disable-line no-undef
+
+chai.Assertion.addChainableMethod('soon',
```

Comment:
Yeah, that's totally fair. The (mostly) common thread was changing things outside the component itself that would cause it to update (like scrolling or inserting HTML into the fixture's light DOM), but not until the next frame, so `updateComplete` isn't necessarily doing anything helpful here, but seemed good to include, except when you consider the potential for misuse.

Anyway, I think we can get rid of it and change these `floating-buttons` test like this:
```javascript
	// fixed with oneEvent()
	it('does not float at bottom of container', async() => {
		const elem = await fixture(floatingButtonsFixture);
		window.scrollTo(0, document.body.scrollHeight);
		await oneEvent(window, 'scroll');
		await expect(elem).to.be.golden();
	});

	// doesn't make sense to me that this one would need a nextFrame(), but it does fix it
	it('does not float when small amount of content', async() => {
		const elem = await fixture(floatingButtonsShortFixture);
		await nextFrame();
		await expect(elem).to.be.golden();
	});

	// fixed with nextFrame()
	it('floats when content added to dom', async() => {
		const elem = await fixture(floatingButtonsShortFixture);
		const contentElem = document.querySelector('#floating-buttons-short-content').querySelector('p');
		contentElem.innerHTML += '<br><br>I love Coffee...';
		await nextFrame();
		await expect(elem).to.be.golden();
	});
```

To be honest I kind of just wanted to get a custom chai property to work 😆

Reply:
Haha cool. Yeah I'm totally fine with tests needing to `await nextFrame()` themselves if they do things that would require waiting for a repaint. At least that's logical.